Adopt consistent `FooSystems` naming convention for system sets

Jondolf opened the pull request:
# Objective

Fixes a part of #14274.

Bevy has an incredibly inconsistent naming convention for its system sets, both internally and across the ecosystem.

<img alt="System sets in Bevy" src="https://github.com/user-attachments/assets/d16e2027-793f-4ba4-9cc9-e780b14a5a1b" width="450" />

*Names of public system set types in Bevy*

Most Bevy types use a naming of `FooSystem` or just `Foo`, but there are also a few `FooSystems` and `FooSet` types. In ecosystem crates on the other hand, `FooSet` is perhaps the most commonly used name in general. Conventions being so wildly inconsistent can make it harder for users to pick names for their own types, to search for system sets on docs.rs, or to even discern which types *are* system sets.

To reign in the inconsistency a bit and help unify the ecosystem, it would be good to establish a common recommended naming convention for system sets in Bevy itself, similar to how plugins are commonly suffixed with `Plugin` (ex: `TimePlugin`). By adopting a consistent naming convention in first-party Bevy, we can softly nudge ecosystem crates to follow suit (for types where it makes sense to do so).

Choosing a naming convention is also relevant now, as the [`bevy_cli` recently adopted lints](https://github.com/TheBevyFlock/bevy_cli/pull/345) to enforce naming for plugins and system sets, and the recommended naming used for system sets is still a bit open.

## Which Name To Use?

Now the contentious part: what naming convention should we actually adopt?

This was discussed on the Bevy Discord at the end of last year, starting [here](<https://discord.com/channels/691052431525675048/692572690833473578/1310659954683936789>). `FooSet` and `FooSystems` were the clear favorites, with `FooSet` very narrowly winning an unofficial poll. However, it seems to me like the consensus was broadly moving towards `FooSystems` at the end and after the poll, with Cart ([source](https://discord.com/channels/691052431525675048/692572690833473578/1311140204974706708)) and later Alice ([source](https://discord.com/channels/691052431525675048/692572690833473578/1311092530732859533)) and also me being in favor of it.

Let's do a quick pros and cons list! Of course these are just what I thought of, so take it with a grain of salt.

`FooSet`:

- Pro: Nice and short!
- Pro: Used by many ecosystem crates.
- Pro: The `Set` suffix comes directly from the trait name `SystemSet`.
- Pro: Pairs nicely with existing APIs like `in_set` and `configure_sets`.
- Con: `Set` by itself doesn't actually indicate that it's related to systems *at all*, apart from the implemented trait. A set of what?
- Con: Is `FooSet` a set of `Foo`s or a system set related to `Foo`? Ex: `ContactSet`, `MeshSet`, `EnemySet`...

`FooSystems`:

- Pro: Very clearly indicates that the type represents a collection of systems. The actual core concept, system(s), is in the name.
- Pro: Parallels nicely with `FooPlugins` for plugin groups.
- Pro: Low risk of conflicts with other names or misunderstandings about what the type is.
- Pro: In most cases, reads *very* nicely and clearly. Ex: `PhysicsSystems` and `AnimationSystems` as opposed to `PhysicsSet` and `AnimationSet`.
- Pro: Easy to search for on docs.rs.
- Con: Usually results in longer names.
- Con: Not yet as widely used.

Really the big problem with `FooSet` is that it doesn't actually describe what it is. It describes what *kind of thing* it is (a set of something), but not *what it is a set of*, unless you know the type or check its docs or implemented traits. `FooSystems` on the other hand is much more self-descriptive in this regard, at the cost of being a bit longer to type.

Ultimately, in some ways it comes down to preference and how you think of system sets. Personally, I was originally in favor of `FooSet`, but have been increasingly on the side of `FooSystems`, especially after seeing what the new names would actually look like in Avian and now Bevy. I prefer it because it usually reads better, is much more clearly related to groups of systems than `FooSet`, and overall *feels* more correct and natural to me in the long term.

For these reasons, and because Alice and Cart also seemed to share a preference for it when it was previously being discussed, I propose that we adopt a `FooSystems` naming convention where applicable.

## Solution

Rename Bevy's system set types to use a consistent `FooSet` naming where applicable.

- `AccessibilitySystem` → `AccessibilitySystems`
- `GizmoRenderSystem` → `GizmoRenderSystems`
- `PickSet` → `PickingSystems`
- `RunFixedMainLoopSystem` → `RunFixedMainLoopSystems`
- `TransformSystem` → `TransformSystems`
- `RemoteSet` → `RemoteSystems`
- `RenderSet` → `RenderSystems`
- `SpriteSystem` → `SpriteSystems`
- `StateTransitionSteps` → `StateTransitionSystems`
- `RenderUiSystem` → `RenderUiSystems`
- `UiSystem` → `UiSystems`
- `Animation` → `AnimationSystems`
- `AssetEvents` → `AssetEventSystems`
- `TrackAssets` → `AssetTrackingSystems`
- `UpdateGizmoMeshes` → `GizmoMeshSystems`
- `InputSystem` → `InputSystems`
- `InputFocusSet` → `InputFocusSystems`
- `ExtractMaterialsSet` → `MaterialExtractionSystems`
- `ExtractMeshesSet` → `MeshExtractionSystems`
- `RumbleSystem` → `RumbleSystems`
- `CameraUpdateSystem` → `CameraUpdateSystems`
- `ExtractAssetsSet` → `AssetExtractionSystems`
- `Update2dText` → `Text2dUpdateSystems`
- `TimeSystem` → `TimeSystems`
- `AudioPlaySet` → `AudioPlaybackSystems`
- `SendEvents` → `EventSenderSystems`
- `EventUpdates` → `EventUpdateSystems`

A lot of the names got slightly longer, but they are also a lot more consistent, and in my opinion the majority of them read much better. For a few of the names I took the liberty of rewording things a bit; definitely open to any further naming improvements.

There are still also cases where the `FooSystems` naming doesn't really make sense, and those I left alone. This primarily includes system sets like `Interned<dyn SystemSet>`, `EnterSchedules<S>`, `ExitSchedules<S>`, or `TransitionSchedules<S>`, where the type has some special purpose and semantics.

## Todo

- [x] Should I keep all the old names as deprecated type aliases? I can do this, but to avoid wasting work I'd prefer to first reach consensus on whether these renames are even desired.
- [x] Migration guide
- [x] Release notes

Jondolf commented:
Slapped M-Needs-Release-Note on this too, since it's a lot of renames, and probably something we want to highlight if we want to try and standardize this in the community a bit more.

janhohenheim reviewed (approved):
Approved not because I carefully read through each file, but because consistency is a good thing :)

Henauxg reviewed (approved):
I noticed the internal renames of
- AmbiguousWithTextSystem → AmbiguousWithText
- AmbiguousWithUpdateText2DLayout → AmbiguousWithUpdateText2dLayout

but I suppose that the `AmbiguousWith` sets are in the "`FooSystems` naming doesn't really make sense" case.

> Should I keep all the old names as deprecated type aliases? 

I'm personally in favor of it. It makes upgrading versions much more pleasant, not needing an external documentation source.

Jondolf commented:
Great, I was honestly expecting this to be more contentious 😅 I'll try to add the deprecated type aliases and migration guide + release notes tomorrow or on Friday

NthTensor reviewed (changes requested):
Remember to write your release notes too. CI has not been configured to block on tags yet, so I'm going to be sending these out in the meantime to make sure nothing with the tag gets merged without them.

Jondolf commented:
Okay, that was kinda grindy but I've added the deprecated type aliases now! Also added the migration guide and release notes, hopefully the format is correct since there weren't any existing notes yet beyond the templates :P

NthTensor reviewed (approved):
Please remove the pr number from the file names, otherwise all correct. Thanks so much for doing this work!

Jondolf commented:
Cool, removed the PR numbers. I assumed I should have it since the [files for 0.16](<https://github.com/bevyengine/bevy-website/tree/main/release-content/0.16/release-notes>) in the bevy_website repo have it, but I guess with the new process it's not needed

NthTensor commented:
With the new process, we want to try to consolidate multiple layered PRs into the same note.

This was probably a huge grind, so thanks again for writing it. In future, if it's helpful, I find it can be good to maintain an ongoing list of changes when doing bulk refactors. It's not that much extra load usually to write down a rename in a second file, and it saves ages of work later.

alice-i-cecile commented:
@Jondolf I'd like to get this in ASAP to avoid more conflicts for you; can you ping me once merge conflicts are resolved?

Jondolf commented:
@alice-i-cecile Done!
